version.h.in

```diff
@@ -19,6 +19,8 @@
 #define @VERSION_TYPE@_VERSION_EXTENDED_STRING  "@@VERSION_TYPE@_VERSION_EXTENDED_STRING@"
 #define @VERSION_TYPE@_VERSION_TWEAK_STRING     "@@VERSION_TYPE@_VERSION_TWEAK_STRING@"
 
+#define @VERSION_TYPE@_GIT_COMMIT_HASH_SHORT    0x@GIT_COMMIT_HASH_SHORT@
```

Comment:
The defines above:
https://github.com/zephyrproject-rtos/zephyr/blob/52be0e393698345fe692eea9d4e317b1b401b541/version.h.in#L12-L13

are also hex values, but the `0x` is prefixed in CMake (and thus not visible in this file).

For consistency we should remove the `0x` here and have CMake prefix the value like it's done for VERSION and VERSION_NUMBER. 
Alternatively update all template code in this file to have `0x` present for hex values.
Mixing styles risks a user looking at this file might assume that only `GIT_COMMIT_HASH_SHORT` has a hex representation.